src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -656,28 +656,42 @@ private async Task<bool> EnumerateFilesFromArtifactsProvider(TContext globalCont
 
                 string filePath = artifact.Uri.GetFilePath();
 
+                long artifactSizeInBytes = -1;
+                try
+                {
+                    artifactSizeInBytes = artifact.SizeInBytes.Value;
```

Comment:
I'm hostile to this change. it doesn't make sense to me that we permit a code path of this kind with a file name that is deemed invalid at runtime. Further, although this code is nicely commented, it's surprising that simply obtaining the size in bytes property provokes this exception.

Reply:
Yeah, I have misgivings, too, but I want to defend this.  For background: Ling's initial PR handled this quite differently.  She had some code that scanned the path for invalid characters before we start doing IOs.  It's not cheap.  For every character that's an invalid filepath character, you scan through the string.  HashSets and similar can get that to what is formally O(n), but with a significant constant overhead.

I definitely don't want to pay that cost during file enumeration.  We could pay it after file enumeration (at higher cost).  But why pay it at all?  Lower layers are going to do this check for us, whether we like it or not.  It's a super rare condition.  So the optimal thing from a perf perspective is to let those layers enforce this for us (probably in optimized native code), and in the extremely rare cases where we have an invalid character in the path: catch and handle the Exception.

The place to deliberately provoke the exception is just before you're sure you need to be interfacing with the filesystem, anyway.  But also as late as possible subject to that constraint because if you're going to screen for file extensions and such, you'd rather do that first.  And you'd like to provoke the Exception in a way that gets you metadata (because cheaper) that you need anyway later.

So, that's how this code wound up in this form and in this place.  Which outcome would you prefer?
* Document that reasoning better in the code, 
* Pay the cost of provoking the Exception earlier at a higher layer (and possibly waste it)
* Do explicit file path scrubbing at a higher layer